libsrc/apple2/videomode.s

```diff
@@ -1,17 +1,51 @@
 ;
 ; Oliver Schmidt, 07.09.2009
 ;
-; unsigned __fastcall__ videomode (unsigned mode);
+; signed char __fastcall__ videomode (unsigned mode);
 ;
-        .ifdef  __APPLE2ENH__
-
         .export         _videomode
 
+        .import         has_80cols_card, returnFFFF
+
+        .ifndef __APPLE2ENH__
+        .constructor    detect_80cols_card, 25
```

Comment:
I think this is a misnomer. You're not detecting anything. You're just relying on the detection performend by ProDOS.